iree/vm/bytecode_dispatch.c

```diff
@@ -1007,19 +1007,19 @@ iree_status_t iree_vm_bytecode_dispatch(
     // Casting and type conversion/emulation
     //===------------------------------------------------------------------===//
 
-#define DISPATCH_OP_CORE_CAST_I32(op_name, src_type, dst_type) \
-  DISPATCH_OP(CORE, op_name, {                                 \
-    int32_t operand = VM_DecOperandRegI32("operand");          \
-    int32_t* result = VM_DecResultRegI32("result");            \
-    *result = (dst_type)((src_type)operand);                   \
+#define DISPATCH_OP_CORE_CAST_I32(op_name, op_func)   \
+  DISPATCH_OP(CORE, op_name, {                        \
+    int32_t operand = VM_DecOperandRegI32("operand"); \
+    int32_t* result = VM_DecResultRegI32("result");   \
+    *result = op_func(operand);                       \
```

Comment:
No need for a separate `DISPATCH_OP_CORE_CAST_I32` macro any longer. As Ben suggested, you can use the already existing `DISPATCH_OP_CORE_UNARY_ALU_I32` macro.

Reply:
After our discussion, we should clean this up in a follow up and define "common" macros as soon as we refactored the dispatcher to use shared implementations.

Reply:
I removed `DISPATCH_OP_CORE_CAST_I32` in favor of `DISPATCH_OP_CORE_UNARY_ALU_I32`. I suggest to reduce the number of macros by unifying the macros, e.g. `DISPATCH_OP_CORE_UNARY_I32` or `DISPATCH_OP_CORE_BINARY_I32`. The less macros - the better :)

Reply:
Thanks for adding a TODO @dajuro. Let's handle cleaning up the macros in a follow up :)